src/jp2image.cpp

```diff
@@ -362,10 +362,9 @@ static void boxes_check(size_t b,size_t m)
                                 long pos = (a == b && (a=='I' || a=='M')) ? 0 : -1;
 
                                 // #1242  Forgive having Exif\0\0 in rawData.pData_
-                                const byte exifHeader[] = { 0x45, 0x78, 0x69, 0x66, 0x00, 0x00 };
-                                for (long i = 0; pos < 0 && i < rawData.size() - static_cast<long>(sizeof(exifHeader));
-                                     i++) {
-                                    if (rawData.cmpBytes(i, exifHeader, sizeof(exifHeader)) == 0)
+                                std::array<byte, 6> exifHeader { 0x45, 0x78, 0x69, 0x66, 0x00, 0x00 };
```

Comment:
Are you considering here to use `auto` as much as possible? I normally like to use `auto` when it makes the code less verbose or even when the type you are using can be infered somehow with the variable name. But I still like to use the specific type when I think it helps to have a better understanding on the operations.

I am open for discussing this in other channel, but I think it would be difficult to reach concensus on this topic with the most active exiv2 contributors

Reply:
I like using auto with types that have constructors. It makes more sense when it looks like a function. It's a moot point honestly.